upstream/failsafe.go

```diff
@@ -41,11 +41,16 @@ func CreateFailSafePolicies(scope Scope, component string, fsCfg *common.Failsaf
 		if err != nil {
 			return nil, err
 		}
-	}
 
-	// For network-level we want the timeout to apply to the overall lifecycle
-	if scope == ScopeNetwork {
-		policies = append(policies, timeoutPolicy)
+		// For network-level we want the timeout to apply to the overall lifecycle
+		if scope == ScopeNetwork {
+			policies = append(policies, timeoutPolicy)
+		}
+
+		// For upstream-level we want the timeout to apply to each individual request towards upstream
+		if scope == ScopeUpstream {
+			policies = append(policies, timeoutPolicy)
+		}
```

Comment:
If policy is moved here it no more "apply to individual requests", is the original code is not working as expected?

Reply:
Oh I think I see original code for upstream-level is using wrong condition, maybe that was the root cause?